src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -274,6 +226,55 @@ public RoutingResponse route(Router router) {
         return itineraries;
     }
 
+    private AccessEgresses getAccessEgresses(Router router, TransitLayer transitLayer) {
+        var accessEgressMapper = new AccessEgressMapper(transitLayer.getStopIndex());
+        var list  = Stream.of(false, true)
+                .parallel()
+                .map(isEgress -> getAccessEgresses(router, accessEgressMapper, isEgress))
+                .collect(Collectors.toList());
+
+        verifyAccessEgress(list.get(0), list.get(1));
+
+        return new AccessEgresses(list.get(0), list.get(1));
+    }
+
+    private Collection<AccessEgress> getAccessEgresses(
+            Router router,
+            AccessEgressMapper accessEgressMapper,
+            boolean isEgress
+    ) {
+        var results = new ArrayList<AccessEgress>();
+        var mode = isEgress ? request.modes.egressMode : request.modes.accessMode;
+
+        // Prepare access/egress lists
+        try (RoutingRequest accessRequest = request.getStreetSearchRequest(mode)) {
+            accessRequest.setRoutingContext(router.graph);
+            if (!isEgress) {
+                accessRequest.allowKeepingRentedBicycleAtDestination = false;
+            }
+
+            var nearbyStops = AccessEgressRouter.streetSearch(
+                    accessRequest,
+                    mode,
+                    isEgress
+            );
+
+            results.addAll(accessEgressMapper.mapNearbyStops(nearbyStops, isEgress));
+
+            // Special handling of flex accesses
+            if (OTPFeature.FlexRouting.isOn() && mode == StreetMode.FLEXIBLE) {
+                var flexAccessList = FlexAccessEgressRouter.routeAccessEgress(
+                        accessRequest,
+                        isEgress
+                );
+
+                results.addAll(accessEgressMapper.mapFlexAccessEgresses(flexAccessList, false));
```

Comment:
```suggestion
                results.addAll(accessEgressMapper.mapFlexAccessEgresses(flexAccessList, isEgress));
```

Reply:
Thanks, I've pushed a commit to correct it.